Fix  Formatting of Copy/Paste of NR Logs

sumitshinde-84 opened the pull request:
## Description

<!-- Describe your changes in detail -->

https://github.com/FlowFuse/flowfuse/assets/110285294/7f39c12b-3734-494d-88e9-c2fa0a594f29

![Screenshot_2023-11-21_22-28-08](https://github.com/FlowFuse/flowfuse/assets/110285294/0329b4df-5880-47ac-ae84-35a8e171b074)


in this change i have use a single span tag to display each item's content. It concatenates the properties like item.date, item.level, item.msg, etc. without creating separate p or div  tag  for each entry. It also includes a br tag to add line breaks between entries, except for the last one. it ensure NR log format will not break and there will be no extra line or space between two rows of log

## Related Issue(s)
Closes #1444 
<!-- What issue does this PR relate to? -->

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

sumitshinde-84 commented:
Hi @joepavitt, could you review this pull request? If any changes are required, please let me know.

sumitshinde-84 commented:
If we want the format to remain as it is, then all entries should be concatenated into a single string. I think that would make it possible.

joepavitt commented:
> If we want the format to remain as it is,

Not sure I follow here what has changed re: the format?

joepavitt commented:
Anything that could be done about the extra line between each copied line do yo think?

sumitshinde-84 commented:
> > If we want the format to remain as it is,
> 
> Not sure I follow here what has changed re: the format?

In this change, rather than using a div to log each value. I have used the p tag to log each value within span (display by default inline), ensuring the values will not appear as different rows when we to try to copy.

sumitshinde-84 commented:
> Anything that could be done about the extra line between each copied line do yo think?

I believe the issue will be resolved if we use a single string for the entire log details

sumitshinde-84 commented:
Hi @joepavitt, could you provide your thoughts on what I have discussed in the #2674 issue section?"

knolleary commented:
@sumitshinde-84 for future PRs, please add a written description to help describe the changes being proposed. The video alone is hard to see what exactly is being changed.

I've had a quick play and can't find an immediate solution to the extra newlines that are appearing between the log lines. It's a definite improvement over what was there before - but would be great to solve this last bit.

sumitshinde-84 commented:
@knolleary  Thanks for reviewing the PR. I completely understand the need for a written description. I'll make sure to include that in the future PRs for better clarity on the proposed changes. I'll continue investigating to find a final solution which will ensure perfect format of NR log for copy-paste.

sumitshinde-84 commented:
Hi @knolleary, I've updated the PR with an improved solution. It ensures that the format won't break when copying and pasting. Also, there won't be any extra lines or spaces between two NR logs. I've added a new video, a screenshot, and a description of the changes. Take a look and let me know if anything else needs adjusting

knolleary commented:
Thanks @sumitshinde-84 

testing it locally, I think some of the whitespacing is a bit off - but I can't put my finger on it exactly. I'll give this a proper look later today.

ZJvandeWeg commented:
@knolleary @sumitshinde-84 Any update on this PR?

sumitshinde-84 commented:
> @knolleary @sumitshinde-84 Any update on this PR?

I am waiting for @knolleary's review. Maybe there are whitespaces somewhere. After receiving the review, I will fix them based on suggestions.

knolleary commented on frontend/src/pages/instance/components/InstanceLogs.vue:
```diff
@@ -10,17 +10,27 @@
+                </template>
+                <span>{{ item.date }}</span>
+                <span>{{ "  " }}</span>
+                <span>[{{ item.level }}]</span>
+                <span v-if="item.level !=='system'">{{ "   " }}</span>
+                <span v-else>{{ " " }}</span>
```
```suggestion
                <span>{{ `[${item.level || ''}]`.padEnd(10, ' ') }}</span>
```

This fixes the whitespace anomaly I saw in the logs.

knolleary reviewed (approved):
Approved pending one minor change to the whitepsace
